File: client/services/employee.service.ts
import { PageResponse } from "@/types/payload/response/common/page.response";
import { ApiClient } from "@/services/config/api.client";
import { Employee } from "@/types/entity/employee";
import { EmployeeRequest } from "@/types/payload/request/employee.request";
import { SuccessResponse } from "@/types/payload/response/common/success.response";

export class EmployeeService {
  static async findAllEmployees(): Promise<PageResponse<Employee[]>> {
    return ApiClient.get("employee").json<PageResponse<Employee[]>>();
  }

  static async addEmployee(
    data: EmployeeRequest,
  ): Promise<SuccessResponse<Employee>> {
    return ApiClient.post("employee", { json: data }).json<
      SuccessResponse<Employee>
    >();
  }
}

File: backend/src/main/java/com/lerneon/backend/services/EmployeeService.java
package com.lerneon.backend.services;

import com.lerneon.backend.models.entity.Employee;
import com.lerneon.backend.models.payload.request.EmployeeRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface EmployeeService {
    Page<Employee> findAllEmployees(Pageable pageable);

    Employee addEmployee(EmployeeRequest employeeRequest);
}
